crates/async-dropper-simple/src/lib.rs

```diff
@@ -55,52 +112,72 @@ impl<T: AsyncDrop + Default + Send + 'static> Drop for AsyncDropper<T> {
-                Some(d) => {
-                    let _ = futures::executor::block_on(tokio::time::timeout(d, task));
-                }
-                None => {
-                    let _ = futures::executor::block_on(task);
-                }
-            });
+            if let Some(time) = timeout {
+                async_scoped::TokioScope::scope_and_block({
+                    |s| {
+                        s.spawn(tokio::time::timeout(time, async {
+                            this.inner.take().expect(INFALLIBLE).async_drop().await;
+                        }));
+                    }
+                });
+            } else {
+                async_scoped::TokioScope::scope_and_block({
+                    |s| {
+                        s.spawn(async {
+                            this.inner.take().expect(INFALLIBLE).async_drop().await;
+                        });
+                    }
+                });
+            }
         }
     }
 }
 
 #[cfg(all(feature = "async-std", not(feature = "tokio")))]
 #[cfg_attr(docsrs, doc(cfg(feature = "async-std")))]
-impl<T: AsyncDrop + Default + Send + 'static> Drop for AsyncDropper<T> {
+impl<T: AsyncDrop + Send> Drop for AsyncDropper<T> {
     fn drop(&mut self) {
         if !self.dropped {
+            // This is the current instance.
+            self.dropped = true;
+            // Grab timeout before mem::take since it replaces self with Default::default() which is None.
+            let timeout = self.timeout;
             let mut this = std::mem::take(self);
-            this.dropped = true;
-
-            // Create task
-            let timeout = self.timeout.clone();
-            let task = async_std::task::spawn(async move {
-                this.inner.async_drop().await;
-            });
-
-            match timeout {
-                Some(d) => {
-                    let _ = futures::executor::block_on(async_std::future::timeout(d, task));
-                }
-                None => {
-                    let _ = futures::executor::block_on(task);
-                }
-            };
+            // This is the Default instance created anew.
+            self.dropped = true;
+
+            if let Some(time) = timeout {
+                async_scoped::AsyncStdScope::scope_and_block({
+                    |s| {
+                        s.spawn(async_std::future::timeout(time, async {
+                            this.inner.take().expect(INFALLIBLE).async_drop().await;
+                        }));
+                    }
+                });
+            } else {
+                async_scoped::AsyncStdScope::scope_and_block({
+                    |s| {
+                        s.spawn(async {
+                            this.inner.take().expect(INFALLIBLE).async_drop().await;
+                        });
+                    }
+                });
+            }
         }
     }
 }
+
+#[cfg(test)]
+#[path = "./__tests__/lib.spec.rs"]
+mod tests;
```

Comment:
Could we keep the tests inline here? 

Also, I'm a bit curious but could you point me at the convention for moving the tests out, and the naming convention for the resulting folders? I'm used to seeing `.spec.x` files or `__tests__` in Ruby and Python codebases and some other languages, I've never seen them used in Rust, curious to know what other projects are doing this!